src/Symfony/Component/HttpClient/Internal/AmpListener.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Internal;
+
+use Amp\Http\Client\Connection\Stream;
+use Amp\Http\Client\EventListener;
+use Amp\Http\Client\Request;
+use Amp\Promise;
+use Amp\Success;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+class AmpListener implements EventListener
+{
+    private $info;
+
+    public function __construct(array &$info)
+    {
+        $info += [
+            'start_time' => 0.0,
+            'connect_time' => 0.0,
+            'pretransfer_time' => 0.0,
+            'starttransfer_time' => 0.0,
+            'total_time' => 0.0,
+            'namelookup_time' => 0.0,
+            'size_upload' => 0, // todo
+            'size_download' => 0, // todo
```

Comment:
@kelunik how should we compute these two metrics? (that's the size before compression, as transferred on the wetwork, it should be updated "live")

Reply:
Resolved by adding an `AmpBody` class.